flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/TableSchema.java

```diff
@@ -244,13 +305,17 @@ public boolean equals(Object o) {
 		}
 		TableSchema schema = (TableSchema) o;
 		return Arrays.equals(fieldNames, schema.fieldNames) &&
-			Arrays.equals(fieldDataTypes, schema.fieldDataTypes);
+			Arrays.equals(fieldDataTypes, schema.fieldDataTypes) &&
+			Arrays.equals(primaryKey, schema.primaryKey) &&
+			Arrays.equals(uniqueKeys, schema.uniqueKeys);
```

Comment:
Arrays.equals doesn't "work" for two dimensional arrays. Use Arrays.deepEquals instead.

Reply:
Good catch! I also change the hashcode to `Arrays.deepHashCode`.